tests/test_tools.py

```diff
@@ -0,0 +1,13 @@
+from functools import partial
+
+from rasterio.features import dataset_features
+from rasterio.tools import JSONSequenceTool
+
+
+def test_dataset_features(tmpdir):
+    """JSON sequence tool works with dataset_features"""
+    tool = JSONSequenceTool(partial(dataset_features, bidx=1, as_mask=True))
+
+    tool("tests/data/RGB.byte.tif", str(tmpdir.join("footprint.jsons")))
```

Comment:
Would be better to assign `tmpdir.join()` to a variable.  Also might be good to use the fixture `path_rgb_byte_tif`

```
out = tmpdir.join("footprint.jsons")
tool(path_rgb_byte_tif, str(out))

assert out.read().count("Feature") == 9
```